californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -679,17 +679,20 @@ private void reject(final RawData raw, final MessageFormatException cause) {
 			rst.setToken(new byte[0]);
 			rst.setDestination(raw.getAddress());
 			rst.setDestinationPort(raw.getPort());
-			for (MessageInterceptor interceptor : interceptors) {
-				interceptor.sendEmptyMessage(rst);
-			}
-			connector.send(serializer.serializeEmptyMessage(rst));
+
+			reject(rst);
```

Comment:
Hard to test: it's only executed, if a MessageFormatException occurs.
But I still believe, we would sent the double rejected message to our own, because we would also double exchange the source/destination.